File: java_SDK/PayHubSDK/src/main/java/com/payhub/ws/api/AuthorizationResponseInformation.java
package com.payhub.ws.api;

import java.io.IOException;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.core.JsonParseException;
import com.fasterxml.jackson.databind.JsonMappingException;
import com.payhub.ws.model.AuthOnlyResponse;

@JsonInclude(JsonInclude.Include.NON_NULL)  
public class AuthorizationResponseInformation {
    private AuthOnlyResponse authOnlyResponse;    
    private Object _links;
    private List<Errors> errors;  
    private String rowData;
    private Object metaData;
    private TransactionManager transactionManager;
    private BillInformation billInformation;
    private CardDataInformation cardDataInformation;
    private CustomerInformation customerInformation;
    private MerchantInformation merchantInformation; 
    
	public String getMetaData() {
		return (String) metaData;
	}
	public void setMetaData(Object metadata) {
		this.metaData = metadata;
	}

	public AuthOnlyResponse getAuthOnlyResponse() {
		return authOnlyResponse;
	}

	public void setAuthOnlyResponse(AuthOnlyResponse authOnlyResponse) {
		this.authOnlyResponse = authOnlyResponse;
	}

	public Object get_links() {
		return _links;
	}

	public void set_links(Object _links) {
		this._links = _links;
	}

	public List<Errors> getErrors() {
		return errors;
	}

	public void setErrors(List<Errors> errors) {
		this.errors = errors;
	}

	public String getRowData() {
		return rowData;
	}

	public void setRowData(String rowData) {
		this.rowData = rowData;
	}
	/**
	 * @param transactionManager the transactionManager to set
	 */
	public void setTransactionManager(TransactionManager transactionManager) {
		this.transactionManager = transactionManager;
	}
	/**
	 * @return the billInformation
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	public BillInformation getBillInformation() throws JsonParseException, JsonMappingException, IOException {
		if(billInformation==null){
			BillInformation b = new BillInformation(this.transactionManager);
			b.setUrl(this.transactionManager.getUrl()+"authonly/");
			b.getBillForSaleInformationByTransactionId(authOnlyResponse.getTransactionId());
			billInformation=b;
		}
		return billInformation;
	}
	/**
	 * @return the cardDataInformation
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	public CardDataInformation getCardDataInformation() throws JsonParseException, JsonMappingException, IOException {
		if(cardDataInformation==null){
			CardDataInformation c = new CardDataInformation(this.transactionManager);
			c.getDataByTransaction(TransactionType.AuthOnly, authOnlyResponse.getTransactionId());
			cardDataInformation=c;			
			}
		return cardDataInformation;
	}

	/**
	 * @return the customerInformation
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	public CustomerInformation getCustomerInformation() throws JsonParseException, JsonMappingException, IOException {
		if(customerInformation==null){
				CustomerInformation c = new CustomerInformation(this.transactionManager);
				c.setUrl(this.transactionManager.getUrl()+"authonly/");
				c.getCustomerForSaleInformationByTransactionId(authOnlyResponse.getTransactionId());
				customerInformation=c;				
			}
		return customerInformation; 
	}

	/**
	 * @return the merchantInformation
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	public MerchantInformation getMerchantInformation() throws JsonParseException, JsonMappingException, IOException {
		if(merchantInformation==null){			
				MerchantInformation m = new MerchantInformation(this.transactionManager);
				m.getDataByTransaction(TransactionType.AuthOnly, authOnlyResponse.getTransactionId());
				merchantInformation=m;				
			}
		return merchantInformation;
	}
}

File: Payhub - SDK/php_SDK/com/payhub/ws/api/AuthorizationResponseInformation.php
<?php 
class AuthorizationResponseInformation {
    public $authOnlyResponse;
    public $_links;
    public $errors;
    public $rowData;
    public $metaData;
    public $transactionManager;
    public $billInformation;
    public $cardDataInformation;
    public $customerInformation;
    public $merchantInformation;

    /**
     * @return mixed
     */
    public function getAuthOnlyResponse()
    {
        return $this->authOnlyResponse;
    }

    /**
     * @param mixed $authOnlyResponse
     */
    public function setAuthOnlyResponse($authOnlyResponse)
    {
        $this->authOnlyResponse=$authOnlyResponse;
    }

    /**
     * @return mixed
     */
    public function getLinks()
    {
        return $this->_links;
    }

    /**
     * @param mixed $links
     */
    public function setLinks($links)
    {
        $this->_links = $links;
    }

    /**
     * @return mixed
     */
    public function getErrors()
    {
        return $this->errors;
    }

    /**
     * @param mixed $errors
     */
    public function setErrors($errors)
    {
        $this->errors = $errors;
    }

    /**
     * @return mixed
     */
    public function getRowData()
    {
        return $this->rowData;
    }

    /**
     * @param mixed $rowData
     */
    public function setRowData($rowData)
    {
        $this->rowData = $rowData;
    }

    /**
     * @return mixed
     */
    public function getMetaData()
    {
        return $this->metaData;
    }

    /**
     * @param mixed $metaData
     */
    public function setMetaData($metaData)
    {
        $this->metaData = $metaData;
    }

    /**
     * @param mixed $transactionManager
     */
    public function setTransactionManager($transactionManager)
    {
        $this->transactionManager = $transactionManager;
    }

    /**
     * @return mixed
     */
    public function getCardDataInformation()
    {
        if($this->cardDataInformation==null){
            $c = new CardDataInformation($this->transactionManager);
            $c.getDataByTransaction(TransactionType::AuthOnly, $this->authOnlyResponse->getTransactionId());
            $this->cardDataInformation=$c;
        }
        return $this->cardDataInformation;
    }

    /**
     * @return mixed
     */
    public function getCustomerInformation()
    {
        if($this->customerInformation==null){
            $c = new CustomerInformation($this->transactionManager);
            $c->setUrl($this->transactionManager->getUrl()."authonly/");
			$c.getDataByTransaction(TransactionType::AuthOnly, $this->authOnlyResponse->getTransactionId());
			$this->customerInformation=$c;
			}
        return $this->customerInformation;
    }

    public function getBillInformation()
    {
        if($this->billInformation==null){
            $b = new BillInformation($this->transactionManager);
            $b->setUrl($this->transactionManager->getUrl()."authonly/");
            $b->getBillForRecurringBillInformationById($this->authOnlyResponse->getTransactionId());
            $this->billInformation=$b;
        }
        return $this->billInformation;
    }

    /**
     * @return mixed
     */
    public function getMerchantInformation()
    {
        if($this->merchantInformation==null){
            $m = new MerchantInformation($this->transactionManager);
            $m.getDataByTransaction(TransactionType::AuthOnly, $this->authOnlyResponse->getTransactionId());
            $merchantInformation=$m;
        }
        return $this->merchantInformation;
    }

    public static function fromArray($data){
        $auth = new AuthorizationResponseInformation();

        foreach ($data as $key => $value){
            if( property_exists( get_class($auth), $key ) ) {
                if($key=="authOnlyResponse"){
                    $auth->{$key}=AuthOnlyResponse::fromArray($value);
                }elseif($key=="errors"){
                    $auth->{$key}=Errors::fromArray($value);
                }
                else{
                    $auth->{$key} = $value;
                }
            }
        }
        return $auth;
    }
} 

?> 